tck/features/CreateAcceptance.feature

```diff
@@ -0,0 +1,607 @@
+      """
+      CREATE (n {id: 12, property: NULL})
+      RETURN n.id AS id
+      """
+    Then the result should be:
+      | id |
+      | 12 |
+    And the side effects should be:
+      | +nodes      | 1 |
+      | +properties | 1 |
+
+  Scenario: Creating a relationship with null properties should not return those properties
+    Given any graph
+    When executing query:
+      """
+      CREATE ()-[r:X {id: 12, property: NULL}]->()
+      RETURN r.id
+      """
+    Then the result should be:
+      | r.id |
+      | 12   |
+    And the side effects should be:
+      | +nodes         | 2 |
+      | +relationships | 1 |
+      | +properties    | 1 |
+
+  Scenario: Create a simple pattern
+    Given any graph
+    When executing query:
+      """
+      CREATE ()-[:R]->()
+      """
+    Then the result should be empty
+    And the side effects should be:
+      | +nodes         | 2 |
+      | +relationships | 1 |
+
+  Scenario: Create a self loop
+    Given any graph
+    When executing query:
+      """
+      CREATE (root:R)-[:LINK]->(root)
+      """
+    Then the result should be empty
+    And the side effects should be:
+      | +nodes         | 1 |
+      | +relationships | 1 |
+      | +labels        | 1 |
+
+  Scenario: Create a self loop using MATCH
+    Given any graph
+    And having executed:
+      """
+      CREATE (:R)
+      """
+    When executing query:
+      """
+      MATCH (root:R)
+      CREATE (root)-[:LINK]->(root)
+      """
+    Then the result should be empty
+    And the side effects should be:
+      | +relationships | 1 |
+
+  Scenario: Create nodes and relationships
+    Given any graph
+    When executing query:
+      """
+      CREATE (a), (b),
+             (a)-[:R]->(b)
+      """
+    Then the result should be empty
+    And the side effects should be:
+      | +nodes         | 2 |
+      | +relationships | 1 |
+
+  Scenario: Create a relationship with a property
+    Given any graph
+    When executing query:
+      """
+      CREATE ()-[:R {prop: 42}]->()
+      """
+    Then the result should be empty
+    And the side effects should be:
+      | +nodes         | 2 |
+      | +relationships | 1 |
+      | +properties    | 1 |
+
+  Scenario: Create a relationship with the correct direction - 1
+    Given any graph
+    And having executed:
+      """
+      CREATE (:X)
+      CREATE (:Y)
+      """
+    When executing query:
+      """
+      MATCH (x:X), (y:Y)
+      CREATE (x)<-[:TYPE]-(y)
+      """
+    Then the result should be empty
+    And the side effects should be:
+      | +relationships | 1 |
+
+  Scenario: Create a relationship with the correct direction - 2
+    Given any graph
+    And having executed:
+      """
+      CREATE (:X)
+      CREATE (:Y)
+      """
+    And having executed:
+      """
+      MATCH (x:X), (y:Y)
+      CREATE (x)<-[:TYPE]-(y)
+      """
+    When executing query:
+      """
+      MATCH (x:X)<-[:TYPE]-(y:Y)
+      RETURN x, y
+      """
```

Comment:
I find that we want to have the query under test in the `When` step, and given that this file is meant to test `CREATE`, I'd say we want a query with `CREATE` in the `When` steps. It seems the `And` step has the create query here, and that the `When` step is meant to make assertions on the result. In #54 I added another `And` step that is meant to be run after the `When` step, to make possible such extra, query-based assertions after an updating query (I call these 'control queries').

Perhaps that's a better approach to this (and subsequent) scenarios? I do like to retain that `When` contains the query that the scenario wants to test.